File: backend/core/manager/ImageManager.py
import dataclasses

from backend.core.utils.type.data.Image import Image
from backend.core.utils.type.data.Stats import Stats
from backend.core.utils.type.data.TimeNode import TimeNode

from typing import TYPE_CHECKING

if (TYPE_CHECKING):
    from core.config.MongoDBConfig import MongoDBConfig

class ImageManager:
    def __init__(self, db: 'MongoDBConfig.MongoDB'):
        self.db = db

    def createImage(self, path: str, **kwargs):
        image = Image(
            path=path,
            tags=kwargs.get('tags', []),
            phenomena=kwargs.get('phenomena', []),
            timeNode=TimeNode(
                acquisitionTime=kwargs.get('acquisitionTime', None),
                analyseTime=kwargs.get('analyseTime', None)
            ),
            stats=Stats(
                satellite=kwargs.get('satellite', None),
                sensor=kwargs.get('sensor', None)
            )
        )

        imageData = dataclasses.asdict(image)

        imageData.pop('_id', None)
        self.db.images.insert_one(imageData)

        return image
    

File: backend/src/core/services/ImageManager.ts
import { Collection, Db } from "mongodb";
import { Image } from "../utils/type/data/Image";
import { CreateImageDto } from "../utils/type/data/CreateImageDto";
import { LogConfig } from "../config/LogConfig";

export class ImageManager {
    public collection: Collection<Image>
    public logger: ReturnType<typeof LogConfig.initLogging>

    constructor(db: Db) {
        this.collection = db.collection<Image>("images")
        this.logger = LogConfig.initLogging()
    }

    public async createImage(data: CreateImageDto): Promise<Image> {

        const newImage: Omit<Image, '_id'> = {
            path: data.path,
            tags: data.tags ?? [],
            phenomena: data.phenomena ?? [],
            timeNode: {
                acquisitionTime: data.acquisitionTime,
                analyseTime: data.analyseTime
            },
            stats: {
                satellite: data.satellite,
                sensor: data.sensor
            }
        }

        const result = await this.collection.insertOne(newImage)

        return {
            _id: result.insertedId,
            ...newImage
        }
    }
}